File: dotnet/Multiplayer/Matchmaking/MatchmakingRequestBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking
{
    public class MatchmakingRequestBase
    {
        public string Request_id { get; set; } = string.Empty;
        public string Matchmaking_id { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public DateTimeOffset Requested_at { get; set; }
        public DateTimeOffset? Responded_at { get; set; }
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingRequestBase.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Base information for a matchmaking join request.
 */
public class MatchmakingRequestBase {
    @JsonProperty("request_id")
    private String requestId;

    @JsonProperty("matchmaking_id")
    private String matchmakingId;

    @JsonProperty("status")
    private String status;

    @JsonProperty("requested_at")
    private String requestedAt;

    @JsonProperty("responded_at")
    private String respondedAt;

    public String getRequestId() {
        return requestId;
    }

    public void setRequestId(String requestId) {
        this.requestId = requestId;
    }

    public String getMatchmakingId() {
        return matchmakingId;
    }

    public void setMatchmakingId(String matchmakingId) {
        this.matchmakingId = matchmakingId;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getRequestedAt() {
        return requestedAt;
    }

    public void setRequestedAt(String requestedAt) {
        this.requestedAt = requestedAt;
    }

    public String getRespondedAt() {
        return respondedAt;
    }

    public void setRespondedAt(String respondedAt) {
        this.respondedAt = respondedAt;
    }
}
